File: src/main/java/ru/mm/surv/capture/service/impl/FilesystemRecordService.java
package ru.mm.surv.capture.service.impl;

import org.apache.commons.io.FilenameUtils;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Service;
import ru.mm.surv.capture.config.FolderConfig;
import ru.mm.surv.capture.service.RecordService;

import java.io.IOException;
import java.io.UncheckedIOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Collection;
import java.util.Collections;
import java.util.stream.Collectors;

@Service
public class FilesystemRecordService implements RecordService {

    private final FolderConfig folders;

    public FilesystemRecordService(FolderConfig folders) {
        this.folders = folders;
    }

    @Override
    public Collection<String> records() {
        var recordsFolder = folders.getMp4();
        if (!Files.exists(recordsFolder)){
            return Collections.emptyList();
        }
        try {
            return Files
                    .list(recordsFolder)
                    .map(Path::getFileName)
                    .map(Path::toString)
                    .map(FilenameUtils::removeExtension)
                    .collect(Collectors.toList());
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }

    @Override
    public Path getMp4File(@NotNull String record) {
        var recordsFolder = folders.getMp4();
        var path = recordsFolder.resolve(record + ".mp4");
        if (!Files.exists(path)) {
            return null;
        }
        return path;
    }

    @Override
    public Path getThumb(@NotNull String record) {
        var thumbsFolder = folders.getMp4Thumb();
        var path = thumbsFolder.resolve(record + ".jpg");
        if (!Files.exists(path)) {
            return null;
        }
        return path;
    }
}

File: src/main/kotlin/ru/mm/surv/capture/service/impl/FilesystemRecordService.kt
package ru.mm.surv.capture.service.impl

import ru.mm.surv.capture.config.FolderConfig
import ru.mm.surv.capture.service.RecordService
import org.apache.commons.io.FilenameUtils
import org.springframework.stereotype.Service
import java.nio.file.Files
import java.nio.file.Path
import kotlin.io.path.deleteIfExists
import kotlin.streams.toList

@Service
class FilesystemRecordService(private val folders: FolderConfig) : RecordService {

    override fun records(): Collection<String> {
        return folders.mp4.takeIf { Files.exists(it) }
            ?.let { Files.list(it) }
            ?.map { it.fileName }
            ?.map(Path::toString)
            ?.map { FilenameUtils.removeExtension(it) }
            ?.sorted(sorter())
            ?.toList()
            ?: emptyList()
    }

    private fun sorter(): Comparator<String> {
        return Comparator { a, b ->
            val dateA = a.split("-", limit = 2)[1]
            val dateB = b.split("-", limit = 2)[1]
            - dateA.compareTo(dateB)
        }
    }

    override fun getMp4File(record: String): Path? {
        return mp4FilePath(record).takeIf { Files.exists(it) }
    }

    override fun getThumb(record: String): Path? {
        return thumbPath(record).takeIf { Files.exists(it) }
    }

    override fun deleteMp4File(record: String) {
        mp4FilePath(record).deleteIfExists()
        thumbPath(record).deleteIfExists()
    }

    private fun thumbPath(record: String) = folders.mp4Thumb.resolve("$record.jpg")

    private fun mp4FilePath(record: String) = folders.mp4.resolve("$record.mp4")
}
